### PR TITLE
[BIOMAGE-1343] Add integration test for adding samples

### PR DESCRIPTION
**Pull request by cosa65**
# Background
#### Link to issue 
https://biomage.atlassian.net/browse/BIOMAGE-1343

#### Links to any Pull Requests related to this
https://github.com/biomage-ltd/ui/pull/504

#### Anything else the reviewers should know about the changes here

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [x] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

**Review by kafkasl — CHANGES_REQUESTED** [review]
I wouldn't add the samples to the repository. We have different options here:
1. Add them using git lfs, but this has a hard limit of 100Mb so I think it's gonna be a problem in the future.
2. Download them from S3.

Both options could be good.

In option 1 we have the 100Mb limit, but to test the upload we may not need more, we can just copy the larger experiments in the github action (as it is done now). Moreover, this means we would be moving data from a bucket to another s3 bucket so that doesn't count towards our aws data transfer quota. 

In option 2 we'd be able to test larger files but we'd have to download them every time which may be inconvenient.

Actually, after writing this I think option 1 is better for the time being.

File: e2e/cypress/support/commands.js [hunk 1]
```diff
@@ -136,6 +143,48 @@ Cypress.Commands.add('deleteMetadata', (metadataTrackName = 'Track 1') => {
   log.end();
 });
 
+// IMPORTANT only works with files that are uncompressed, gem2s fails with compressed files
+// (probably due to file sizes wrong calculation, if we consider this should be fixed
+// the work should probably be done in the forked repo cypress-file-upload)
```
[the remark this answers is not in the document]
Comment (cosa65) [reply e2e/cypress/support/commands.js:148]: Yes, if I used `matrix.mtx.gz` for example gem2s fails while if I gunzip it and then use `matrix.mtx` it works correctly, I believe it could be related to the sizes of gzipped files when loaded on cypress don't really match the real file size.

Here's a ticket for fixing this: [ticket](https://biomage.atlassian.net/browse/BIOMAGE-1435).

I don't think this ticket is very high priority however so I set it to low since it doesn't block us from testing anything but the upload of gzipped files.

File: e2e/cypress/integration/ui-smoke/ui-add-sample.spec.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    return true;
+  });
+
+  it('Adds a new sample correctly by drag and drop', () => {
+    cy.selectProject(projectName);
+
+    cy.addSample(addFileActions.DRAG_AND_DROP);
+
+    // Sample cell shows up
+    cy.get('.data-test-sample-name-cell').should('be.visible');
```
[the remark this answers is not in the document]
Comment (cosa65) [reply e2e/cypress/integration/ui-smoke/ui-add-sample.spec.js:38]: I you look at `SamplesTable.jsx`, each of the rendered components inside the table is called `cell`, do you think I should call it `data-test-sample-name-table-cell` instead or something like that to avoid confusion?

File: e2e/cypress/support/commands.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (cosa65) [reply e2e/cypress/support/commands.js:148]: Agreed on the lfs thing, I'll try it out then.

File: e2e/cypress/integration/ui-smoke/ui-add-sample.spec.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (aerlaut) [reply e2e/cypress/integration/ui-smoke/ui-add-sample.spec.js:38]: Maybe change to `data-test-samples-table-name-cell`?

File: e2e/cypress/integration/ui-smoke/ui-create-delete-metadata.spec.js [hunk 3]
```diff
@@ -10,21 +10,14 @@ describe('Adds metadata to a sample in a created project', () => {
...
       },
     ).as('putProject');
 
-    cy.intercept(
-      {
-        method: 'DELETE',
-        url: '*/projects/*',
-      },
-    ).as('deleteProject');
-
```
Comment (aerlaut) [inline e2e/cypress/integration/ui-smoke/ui-create-delete-metadata.spec.js:27]: Why is this deleted?

File: e2e/cypress/integration/ui-smoke/ui-add-sample.spec.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (kafkasl) [reply e2e/cypress/integration/ui-smoke/ui-add-sample.spec.js:38]: I think the cell part is just a particular implementation name for the table but it doesn't give any extra info to the outside. I would just call it '.data-test-sample-name'. Otherwise, it is too long and verbose. If there are other places where this name can be used (thus having potential conflicts) then I would resort to more complex selectors instead of longer names e.g. `cy.get('.data-test-samples-table > .data-test-sample-name')`. imho it's easier to see the structure and it becomes more readable instead of trying to put all the hierarchy structure in the name as agi suggested.

Regarding the actual selection, shouldn't it select the elements matching the given sample name instead of just all ay of them?  Something like (syntax might be incorrect):

`cy.get('.data-test-sample-name-cell').contains(sampleName).should('be.visible');`

[the remark this answers is not in the document]
Comment (cosa65) [reply e2e/cypress/integration/ui-smoke/ui-add-sample.spec.js:38]: Im going with `data-test-samples-in-table` which seems descriptive and I think takes into account the suggestions in thecomments

File: e2e/cypress/integration/ui-smoke/ui-create-delete-metadata.spec.js (hunk 3 above)
Comment (cosa65) [reply e2e/cypress/integration/ui-smoke/ui-create-delete-metadata.spec.js:27]: Because we aren't using the delete method at any point in this test, we only care about the update calls

**Review by kafkasl — APPROVED** [review]
